# 최적화된 GPTQ 양자화 (베이스라인 기반 개선)

## 조사 결과 기반 최적화

### 참고 자료
- [vLLM Quantization Guide](https://docs.vllm.ai/en/latest/features/quantization/)
- [LLM Compressor GPTQModifier](https://docs.vllm.ai/projects/llm-compressor/en/stable/reference/llmcompressor/modifiers/quantization/gptq/)
- [vLLM Quantization Benchmarks](https://docs.jarvislabs.ai/blog/vllm-quantization-complete-guide-benchmarks)

### 베이스라인 대비 변경점

| 항목 | 베이스라인 | 최적화 버전 | 효과 |
|------|-----------|------------|------|
| actorder | (미지정) | **weight** | 정확도 ~2점 향상 |
| block_size | (미지정) | **128** | Marlin 커널 호환 |
| dampening_frac | (미지정) | **0.001** | Hessian 안정화 |
| symmetric | (미지정) | **true** | 추론 속도 향상 |

### 핵심 발견
1. **actorder="weight"**: W4A16에서 정확도 최대 2점 향상 (llmcompressor v0.8.0+)
2. **Marlin 커널**: vLLM이 자동 적용, GPTQ 2.6배 속도 향상
3. **group_size=128**: Marlin 호환 필수 조건

---

# 1. Import

In [1]:
import os
import torch
import shutil
from pathlib import Path

from datasets import load_dataset
from transformers import AutoModelForCausalLM, AutoTokenizer

from llmcompressor import oneshot
from llmcompressor.modifiers.quantization import GPTQModifier

print(f"PyTorch: {torch.__version__}")
print(f"CUDA: {torch.cuda.is_available()}")
if torch.cuda.is_available():
    print(f"GPU: {torch.cuda.get_device_name(0)}")
else:
    print("⚠️ CPU 모드로 실행됩니다")
print("\n✅ Import 완료!")

PyTorch: 2.9.1
CUDA: False
⚠️ CPU 모드로 실행됩니다

✅ Import 완료!


# 2. 설정 (베이스라인 + 최적화)

In [2]:
# ============================================================================
# 모델 설정 (베이스라인 동일)
# ============================================================================
MODEL_ID = "./open/base_model"  # 로컬 모델
OUT_DIR = "./model"             # 제출용 폴더명

# ============================================================================
# 데이터셋 설정 (베이스라인 동일)
# ============================================================================
DATASET_ID = "LGAI-EXAONE/MANTA-1M"
DATASET_SPLIT = "train"

# ============================================================================
# 캘리브레이션 설정 (베이스라인 동일)
# ============================================================================
NUM_CALIBRATION_SAMPLES = 256
MAX_SEQUENCE_LENGTH = 512

# ============================================================================
# 양자화 설정 (베이스라인 동일)
# ============================================================================
SCHEME = "W4A16"
TARGETS = ["Linear"]
IGNORE = ["embed_tokens", "lm_head"]

# ============================================================================
# ⭐ 최적화 설정 (NEW!)
# ============================================================================
# 참고: https://docs.vllm.ai/projects/llm-compressor/en/stable/reference/llmcompressor/modifiers/quantization/gptq/

BLOCK_SIZE = 128        # Marlin 커널 호환 (필수)
DAMPENING_FRAC = 0.001  # Hessian 안정화 (기본값)
ACTORDER = "weight"     # W4A16 정확도 향상 (v0.8.0+ 권장)

# 원본 모델 크기
ORIGINAL_MODEL_SIZE_GB = 2.56

print("=" * 60)
print("최적화된 GPTQ 설정")
print("=" * 60)
print(f"MODEL_ID: {MODEL_ID}")
print(f"SCHEME: {SCHEME}")
print(f"SAMPLES: {NUM_CALIBRATION_SAMPLES}")
print(f"MAX_LEN: {MAX_SEQUENCE_LENGTH}")
print("---")
print("⭐ 최적화 파라미터:")
print(f"  BLOCK_SIZE: {BLOCK_SIZE} (Marlin 호환)")
print(f"  DAMPENING_FRAC: {DAMPENING_FRAC}")
print(f"  ACTORDER: {ACTORDER} (정확도 향상)")
print("=" * 60)

최적화된 GPTQ 설정
MODEL_ID: ./open/base_model
SCHEME: W4A16
SAMPLES: 256
MAX_LEN: 512
---
⭐ 최적화 파라미터:
  BLOCK_SIZE: 128 (Marlin 호환)
  DAMPENING_FRAC: 0.001
  ACTORDER: weight (정확도 향상)


# 3. 모델 로드

In [3]:
print("[INFO] 모델 로드 중...")

tokenizer = AutoTokenizer.from_pretrained(
    MODEL_ID,
    trust_remote_code=True,
)

# GPU/CPU 자동 선택
if torch.cuda.is_available():
    model = AutoModelForCausalLM.from_pretrained(
        MODEL_ID,
        torch_dtype=torch.bfloat16,
        trust_remote_code=True,
    )
else:
    model = AutoModelForCausalLM.from_pretrained(
        MODEL_ID,
        torch_dtype=torch.float32,
        trust_remote_code=True,
        device_map="cpu",
    )

print(f"[INFO] 모델 파라미터: {model.num_parameters():,}")
print("[INFO] 모델/토크나이저 로드 완료")

`torch_dtype` is deprecated! Use `dtype` instead!


[INFO] 모델 로드 중...
[INFO] 모델 파라미터: 1,279,391,488
[INFO] 모델/토크나이저 로드 완료


# 4. 데이터셋 로드

In [4]:
print("[INFO] 캘리브레이션 데이터 로드 중...")

ds = load_dataset(
    DATASET_ID,
    split=f"{DATASET_SPLIT}[:{NUM_CALIBRATION_SAMPLES}]",
)

def preprocess(example):
    return {
        "text": tokenizer.apply_chat_template(
            example["conversations"],
            add_generation_prompt=True,
            tokenize=False
        )
    }

ds = ds.map(preprocess)

print(f"[INFO] 데이터셋 크기: {len(ds)}")
print("[INFO] 데이터 전처리 완료")

[INFO] 캘리브레이션 데이터 로드 중...
[INFO] 데이터셋 크기: 256
[INFO] 데이터 전처리 완료


# 5. GPTQ 양자화 (최적화 버전)

### 베이스라인 vs 최적화

```python
# 베이스라인
GPTQModifier(
    scheme=SCHEME,
    targets=TARGETS,
    ignore=IGNORE,
)

# 최적화 (아래)
GPTQModifier(
    scheme=SCHEME,
    targets=TARGETS,
    ignore=IGNORE,
    block_size=128,       # ⭐ Marlin 호환
    dampening_frac=0.001, # ⭐ Hessian 안정화
    actorder="weight",    # ⭐ 정확도 향상
)
```

In [5]:
print("[INFO] GPTQ 양자화 시작 (최적화 버전)")
print(f"  - scheme: {SCHEME}")
print(f"  - samples: {NUM_CALIBRATION_SAMPLES}")
print(f"  - max_len: {MAX_SEQUENCE_LENGTH}")
print(f"  - block_size: {BLOCK_SIZE} (Marlin 호환)")
print(f"  - actorder: {ACTORDER} (정확도 향상)")
print(f"  - dampening_frac: {DAMPENING_FRAC}")

if torch.cuda.is_available():
    print("\n🚀 GPU 모드: 10-20분 예상\n")
else:
    print("\n⏳ CPU 모드: 15-30분 예상\n")

# ============================================================================
# 최적화된 GPTQ 레시피
# ============================================================================
recipe = [
    GPTQModifier(
        scheme=SCHEME,
        targets=TARGETS,
        ignore=IGNORE,
        # ⭐ 최적화 파라미터
        block_size=BLOCK_SIZE,        # Marlin 커널 호환
        dampening_frac=DAMPENING_FRAC, # Hessian 안정화
        actorder=ACTORDER,            # 가중치 기반 정렬 (정확도↑)
    )
]

oneshot(
    model=model,
    dataset=ds,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
)

print("\n[INFO] GPTQ 양자화 완료!")

[INFO] GPTQ 양자화 시작 (최적화 버전)
  - scheme: W4A16
  - samples: 256
  - max_len: 512
  - block_size: 128 (Marlin 호환)
  - actorder: weight (정확도 향상)
  - dampening_frac: 0.001

⏳ CPU 모드: 15-30분 예상



Tokenizing:   0%|          | 0/256 [00:00<?, ? examples/s]

2026-02-10T14:31:23.547322+0900 | reset | INFO - Compression lifecycle reset
2026-02-10T14:31:23.549407+0900 | from_modifiers | INFO - Creating recipe from modifiers
2026-02-10T14:31:23.578409+0900 | initialize | INFO - Compression lifecycle initialized for 1 modifiers
2026-02-10T14:31:23.579022+0900 | IndependentPipeline | INFO - Inferred `SequentialPipeline` for `GPTQModifier`
2026-02-10T14:31:23.586487+0900 | dispatch_for_sequential | WARNING - CUDA/XPU is not available! Compressing model on CPU instead


W0210 14:31:23.619000 77652 torch/fx/_symbolic_trace.py:52] is_fx_tracing will return true for both fx.symbolic_trace and torch.export. Please use is_fx_tracing_symbolic_tracing() for specifically fx.symbolic_trace or torch.compiler.is_compiling() for specifically torch.export/compile.
(1/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:05<00:00,  3.91it/s]

2026-02-10T14:32:29.366854+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.q_proj using 256 samples


2026-02-10T14:32:29.830866+0900 | compress | METRIC - time 0.46s
2026-02-10T14:32:29.831415+0900 | compress | METRIC - error 1.07
2026-02-10T14:32:29.835936+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:32:29.836433+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:32:29.838512+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.k_proj using 256 samples
2026-02-10T14:32:30.118041+0900 | compress | METRIC - time 0.28s
2026-02-10T14:32:30.118576+0900 | compress | METRIC - error 0.31
2026-02-10T14:32:30.119723+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:32:30.120106+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:32:30.121709+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.v_proj using 256 samples
2026-02-10T14:32:30.393797+0900 | compress | METRIC - time 0.27s
2026-02-10T14:32:30.39

(2/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:02<00:00,  4.11it/s]

2026-02-10T14:33:47.582541+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.q_proj using 256 samples


2026-02-10T14:33:48.121994+0900 | compress | METRIC - time 0.54s
2026-02-10T14:33:48.122615+0900 | compress | METRIC - error 4.53
2026-02-10T14:33:48.124881+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:33:48.125258+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:33:48.127093+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.k_proj using 256 samples
2026-02-10T14:33:48.373489+0900 | compress | METRIC - time 0.25s
2026-02-10T14:33:48.373955+0900 | compress | METRIC - error 1.29
2026-02-10T14:33:48.375072+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:33:48.375416+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:33:48.376236+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.v_proj using 256 samples
2026-02-10T14:33:48.624517+0900 | compress | METRIC - time 0.25s
2026-02-10T14:33:48.62

(3/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.33it/s]

2026-02-10T14:35:02.149063+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.q_proj using 256 samples


2026-02-10T14:35:02.576203+0900 | compress | METRIC - time 0.43s
2026-02-10T14:35:02.576760+0900 | compress | METRIC - error 12.51
2026-02-10T14:35:02.579418+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:35:02.579917+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:35:02.582061+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.k_proj using 256 samples
2026-02-10T14:35:02.875053+0900 | compress | METRIC - time 0.29s
2026-02-10T14:35:02.875535+0900 | compress | METRIC - error 3.51
2026-02-10T14:35:02.876661+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:35:02.877028+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:35:02.878120+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.v_proj using 256 samples
2026-02-10T14:35:03.133759+0900 | compress | METRIC - time 0.26s
2026-02-10T14:35:03.1

(4/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.20it/s]

2026-02-10T14:36:18.470914+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.q_proj using 256 samples


2026-02-10T14:36:18.889345+0900 | compress | METRIC - time 0.42s
2026-02-10T14:36:18.889829+0900 | compress | METRIC - error 25.72
2026-02-10T14:36:18.891641+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:36:18.891916+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:36:18.893822+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.k_proj using 256 samples
2026-02-10T14:36:19.173914+0900 | compress | METRIC - time 0.28s
2026-02-10T14:36:19.174394+0900 | compress | METRIC - error 7.25
2026-02-10T14:36:19.175527+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:36:19.175787+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:36:19.176712+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.v_proj using 256 samples
2026-02-10T14:36:19.437644+0900 | compress | METRIC - time 0.26s
2026-02-10T14:36:19.4

(5/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.22it/s]

2026-02-10T14:37:34.573232+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.q_proj using 256 samples


2026-02-10T14:37:35.112426+0900 | compress | METRIC - time 0.54s
2026-02-10T14:37:35.113168+0900 | compress | METRIC - error 48.94
2026-02-10T14:37:35.115962+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:37:35.117070+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:37:35.119790+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.k_proj using 256 samples
2026-02-10T14:37:35.447221+0900 | compress | METRIC - time 0.33s
2026-02-10T14:37:35.447737+0900 | compress | METRIC - error 13.56
2026-02-10T14:37:35.448935+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:37:35.449259+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:37:35.450115+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.v_proj using 256 samples
2026-02-10T14:37:35.717545+0900 | compress | METRIC - time 0.27s
2026-02-10T14:37:35.

(6/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.22it/s]

2026-02-10T14:38:50.724056+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.q_proj using 256 samples


2026-02-10T14:38:51.854068+0900 | compress | METRIC - time 1.13s
2026-02-10T14:38:51.854639+0900 | compress | METRIC - error 79.38
2026-02-10T14:38:51.855810+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:38:51.856162+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:38:51.857946+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.k_proj using 256 samples
2026-02-10T14:38:52.506219+0900 | compress | METRIC - time 0.65s
2026-02-10T14:38:52.506803+0900 | compress | METRIC - error 23.35
2026-02-10T14:38:52.508092+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:38:52.508489+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:38:52.509443+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.v_proj using 256 samples
2026-02-10T14:38:52.852385+0900 | compress | METRIC - time 0.34s
2026-02-10T14:38:52.

(7/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:10<00:00,  3.65it/s]

2026-02-10T14:40:21.936287+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.q_proj using 256 samples


2026-02-10T14:40:22.568075+0900 | compress | METRIC - time 0.63s
2026-02-10T14:40:22.568644+0900 | compress | METRIC - error 115.21
2026-02-10T14:40:22.572719+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:40:22.573373+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:40:22.575726+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.k_proj using 256 samples
2026-02-10T14:40:22.980075+0900 | compress | METRIC - time 0.40s
2026-02-10T14:40:22.980604+0900 | compress | METRIC - error 31.71
2026-02-10T14:40:22.981827+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:40:22.982229+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:40:22.983362+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.v_proj using 256 samples
2026-02-10T14:40:23.484111+0900 | compress | METRIC - time 0.50s
2026-02-10T14:40:23

(8/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:01<00:00,  4.18it/s]

2026-02-10T14:41:50.441882+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.q_proj using 256 samples


2026-02-10T14:41:50.890748+0900 | compress | METRIC - time 0.45s
2026-02-10T14:41:50.891456+0900 | compress | METRIC - error 173.75
2026-02-10T14:41:50.896762+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:41:50.897384+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:41:50.899838+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.k_proj using 256 samples
2026-02-10T14:41:51.229957+0900 | compress | METRIC - time 0.33s
2026-02-10T14:41:51.230486+0900 | compress | METRIC - error 48.87
2026-02-10T14:41:51.231622+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:41:51.231983+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:41:51.232939+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.v_proj using 256 samples
2026-02-10T14:41:51.499680+0900 | compress | METRIC - time 0.27s
2026-02-10T14:41:51

(9/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:58<00:00,  4.39it/s]

2026-02-10T14:43:04.309271+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.q_proj using 256 samples


2026-02-10T14:43:04.712991+0900 | compress | METRIC - time 0.40s
2026-02-10T14:43:04.713551+0900 | compress | METRIC - error 190.15
2026-02-10T14:43:04.714926+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:43:04.715303+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:43:04.717623+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.k_proj using 256 samples
2026-02-10T14:43:04.971903+0900 | compress | METRIC - time 0.25s
2026-02-10T14:43:04.972431+0900 | compress | METRIC - error 54.23
2026-02-10T14:43:04.973594+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:43:04.974013+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:43:04.975047+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.v_proj using 256 samples
2026-02-10T14:43:05.226113+0900 | compress | METRIC - time 0.25s
2026-02-10T14:43:05

(10/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.26it/s]

2026-02-10T14:44:19.331269+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.q_proj using 256 samples


2026-02-10T14:44:19.778613+0900 | compress | METRIC - time 0.45s
2026-02-10T14:44:19.779120+0900 | compress | METRIC - error 253.19
2026-02-10T14:44:19.782621+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:44:19.783075+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:44:19.784935+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.k_proj using 256 samples
2026-02-10T14:44:20.048129+0900 | compress | METRIC - time 0.26s
2026-02-10T14:44:20.048663+0900 | compress | METRIC - error 74.66
2026-02-10T14:44:20.049681+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:44:20.049979+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:44:20.050915+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.v_proj using 256 samples
2026-02-10T14:44:20.310470+0900 | compress | METRIC - time 0.26s
2026-02-10T14:44:20

(11/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.21it/s]

2026-02-10T14:45:35.514737+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.q_proj using 256 samples


2026-02-10T14:45:36.020125+0900 | compress | METRIC - time 0.50s
2026-02-10T14:45:36.020740+0900 | compress | METRIC - error 275.44
2026-02-10T14:45:36.022016+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:45:36.022363+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:45:36.024382+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.k_proj using 256 samples
2026-02-10T14:45:36.290009+0900 | compress | METRIC - time 0.27s
2026-02-10T14:45:36.290821+0900 | compress | METRIC - error 74.06
2026-02-10T14:45:36.292156+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:45:36.292516+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:45:36.293747+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.v_proj using 256 samples
2026-02-10T14:45:36.631559+0900 | compress | METRIC - time 0.34s
2026-02-10T14:45:

(12/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:01<00:00,  4.18it/s]

2026-02-10T14:46:51.944017+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.q_proj using 256 samples


2026-02-10T14:46:52.402786+0900 | compress | METRIC - time 0.46s
2026-02-10T14:46:52.403433+0900 | compress | METRIC - error 298.80
2026-02-10T14:46:52.406356+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:46:52.406963+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:46:52.409014+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.k_proj using 256 samples
2026-02-10T14:46:52.666952+0900 | compress | METRIC - time 0.26s
2026-02-10T14:46:52.667406+0900 | compress | METRIC - error 84.51
2026-02-10T14:46:52.668383+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:46:52.668670+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:46:52.669486+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.v_proj using 256 samples
2026-02-10T14:46:53.020242+0900 | compress | METRIC - time 0.35s
2026-02-10T14:46:

(13/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.26it/s]

2026-02-10T14:48:07.964226+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.q_proj using 256 samples


2026-02-10T14:48:08.387869+0900 | compress | METRIC - time 0.42s
2026-02-10T14:48:08.388382+0900 | compress | METRIC - error 334.67
2026-02-10T14:48:08.389517+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:48:08.389822+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:48:08.391640+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.k_proj using 256 samples
2026-02-10T14:48:08.650290+0900 | compress | METRIC - time 0.26s
2026-02-10T14:48:08.650730+0900 | compress | METRIC - error 91.74
2026-02-10T14:48:08.651707+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:48:08.652001+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:48:08.652855+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.v_proj using 256 samples
2026-02-10T14:48:08.917760+0900 | compress | METRIC - time 0.26s
2026-02-10T14:48:

(14/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:03<00:00,  4.04it/s]

2026-02-10T14:49:26.552503+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.q_proj using 256 samples


2026-02-10T14:49:27.033676+0900 | compress | METRIC - time 0.48s
2026-02-10T14:49:27.034160+0900 | compress | METRIC - error 374.68
2026-02-10T14:49:27.036628+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:49:27.037001+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:49:27.038943+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.k_proj using 256 samples
2026-02-10T14:49:27.306409+0900 | compress | METRIC - time 0.27s
2026-02-10T14:49:27.306901+0900 | compress | METRIC - error 104.86
2026-02-10T14:49:27.307912+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:49:27.308238+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:49:27.309139+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.v_proj using 256 samples
2026-02-10T14:49:27.567052+0900 | compress | METRIC - time 0.26s
2026-02-10T14:49

(15/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.27it/s]

2026-02-10T14:50:41.718686+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.q_proj using 256 samples


2026-02-10T14:50:42.178586+0900 | compress | METRIC - time 0.46s
2026-02-10T14:50:42.179112+0900 | compress | METRIC - error 406.71
2026-02-10T14:50:42.181854+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:50:42.182473+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:50:42.184571+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.k_proj using 256 samples
2026-02-10T14:50:42.463451+0900 | compress | METRIC - time 0.28s
2026-02-10T14:50:42.463969+0900 | compress | METRIC - error 121.86
2026-02-10T14:50:42.465092+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:50:42.465409+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:50:42.466336+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.v_proj using 256 samples
2026-02-10T14:50:42.811051+0900 | compress | METRIC - time 0.34s
2026-02-10T14:50

(16/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.27it/s]

2026-02-10T14:51:58.001362+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.q_proj using 256 samples


2026-02-10T14:51:58.464708+0900 | compress | METRIC - time 0.46s
2026-02-10T14:51:58.465322+0900 | compress | METRIC - error 421.56
2026-02-10T14:51:58.468366+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:51:58.469009+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:51:58.471240+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.k_proj using 256 samples
2026-02-10T14:51:58.744235+0900 | compress | METRIC - time 0.27s
2026-02-10T14:51:58.744794+0900 | compress | METRIC - error 118.52
2026-02-10T14:51:58.745892+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:51:58.746262+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:51:58.747285+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.v_proj using 256 samples
2026-02-10T14:51:59.006998+0900 | compress | METRIC - time 0.26s
2026-02-10T14:51

(17/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.25it/s]

2026-02-10T14:53:13.581531+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.q_proj using 256 samples


2026-02-10T14:53:14.018946+0900 | compress | METRIC - time 0.44s
2026-02-10T14:53:14.019652+0900 | compress | METRIC - error 498.89
2026-02-10T14:53:14.020793+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:53:14.021143+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:53:14.022978+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.k_proj using 256 samples
2026-02-10T14:53:14.282579+0900 | compress | METRIC - time 0.26s
2026-02-10T14:53:14.283163+0900 | compress | METRIC - error 130.72
2026-02-10T14:53:14.284133+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:53:14.284422+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:53:14.285388+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.v_proj using 256 samples
2026-02-10T14:53:14.541903+0900 | compress | METRIC - time 0.26s
2026-02-10T14:53

(18/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:01<00:00,  4.14it/s]

2026-02-10T14:54:34.863534+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.q_proj using 256 samples


2026-02-10T14:54:35.319513+0900 | compress | METRIC - time 0.45s
2026-02-10T14:54:35.320052+0900 | compress | METRIC - error 516.96
2026-02-10T14:54:35.322809+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:54:35.323378+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:54:35.325336+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.k_proj using 256 samples
2026-02-10T14:54:35.578319+0900 | compress | METRIC - time 0.25s
2026-02-10T14:54:35.578757+0900 | compress | METRIC - error 140.25
2026-02-10T14:54:35.579873+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:54:35.580271+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:54:35.581259+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.v_proj using 256 samples
2026-02-10T14:54:35.860001+0900 | compress | METRIC - time 0.28s
2026-02-10T14:54

(19/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.24it/s]

2026-02-10T14:55:51.081072+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.q_proj using 256 samples


2026-02-10T14:55:51.502290+0900 | compress | METRIC - time 0.42s
2026-02-10T14:55:51.502767+0900 | compress | METRIC - error 568.79
2026-02-10T14:55:51.504377+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:55:51.504669+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:55:51.506541+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.k_proj using 256 samples
2026-02-10T14:55:51.765605+0900 | compress | METRIC - time 0.26s
2026-02-10T14:55:51.766083+0900 | compress | METRIC - error 161.61
2026-02-10T14:55:51.767165+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:55:51.767471+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:55:51.768275+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.v_proj using 256 samples
2026-02-10T14:55:52.025232+0900 | compress | METRIC - time 0.26s
2026-02-10T14:55

(20/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:02<00:00,  4.07it/s]

2026-02-10T14:57:08.884912+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.q_proj using 256 samples


2026-02-10T14:57:09.389108+0900 | compress | METRIC - time 0.50s
2026-02-10T14:57:09.389707+0900 | compress | METRIC - error 570.65
2026-02-10T14:57:09.391455+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:57:09.391834+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:57:09.394341+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.k_proj using 256 samples
2026-02-10T14:57:09.670414+0900 | compress | METRIC - time 0.28s
2026-02-10T14:57:09.670963+0900 | compress | METRIC - error 162.73
2026-02-10T14:57:09.672108+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:57:09.672412+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:57:09.673418+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.v_proj using 256 samples
2026-02-10T14:57:09.967858+0900 | compress | METRIC - time 0.29s
2026-02-10T14:57

(21/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.26it/s]

2026-02-10T14:58:24.777916+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.q_proj using 256 samples


2026-02-10T14:58:25.208652+0900 | compress | METRIC - time 0.43s
2026-02-10T14:58:25.209250+0900 | compress | METRIC - error 677.17
2026-02-10T14:58:25.210556+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:58:25.210876+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:58:25.212967+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.k_proj using 256 samples
2026-02-10T14:58:25.500966+0900 | compress | METRIC - time 0.29s
2026-02-10T14:58:25.501560+0900 | compress | METRIC - error 181.05
2026-02-10T14:58:25.503040+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:58:25.503476+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:58:25.504744+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.v_proj using 256 samples
2026-02-10T14:58:25.848229+0900 | compress | METRIC - time 0.34s
2026-02-10T14:58

(22/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.28it/s]

2026-02-10T14:59:40.496493+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.q_proj using 256 samples


2026-02-10T14:59:41.002384+0900 | compress | METRIC - time 0.50s
2026-02-10T14:59:41.002953+0900 | compress | METRIC - error 777.56
2026-02-10T14:59:41.005817+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:59:41.006305+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T14:59:41.008467+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.k_proj using 256 samples
2026-02-10T14:59:41.269656+0900 | compress | METRIC - time 0.26s
2026-02-10T14:59:41.270217+0900 | compress | METRIC - error 207.92
2026-02-10T14:59:41.271362+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T14:59:41.271680+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T14:59:41.272720+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.v_proj using 256 samples
2026-02-10T14:59:41.531723+0900 | compress | METRIC - time 0.26s
2026-02-10T14:59

(23/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.29it/s]

2026-02-10T15:00:55.969193+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.q_proj using 256 samples


2026-02-10T15:00:56.422433+0900 | compress | METRIC - time 0.45s
2026-02-10T15:00:56.422961+0900 | compress | METRIC - error 849.91
2026-02-10T15:00:56.425846+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:00:56.426358+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:00:56.428386+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.k_proj using 256 samples
2026-02-10T15:00:56.690650+0900 | compress | METRIC - time 0.26s
2026-02-10T15:00:56.691155+0900 | compress | METRIC - error 240.42
2026-02-10T15:00:56.692248+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:00:56.692584+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:00:56.693498+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.v_proj using 256 samples
2026-02-10T15:00:56.956896+0900 | compress | METRIC - time 0.26s
2026-02-10T15:00

(24/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.26it/s]

2026-02-10T15:02:11.480518+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.q_proj using 256 samples


2026-02-10T15:02:11.961855+0900 | compress | METRIC - time 0.48s
2026-02-10T15:02:11.962368+0900 | compress | METRIC - error 944.17
2026-02-10T15:02:11.963843+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:02:11.964225+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:02:11.966272+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.k_proj using 256 samples
2026-02-10T15:02:12.253692+0900 | compress | METRIC - time 0.29s
2026-02-10T15:02:12.254211+0900 | compress | METRIC - error 277.75
2026-02-10T15:02:12.255299+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:02:12.255592+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:02:12.256452+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.v_proj using 256 samples
2026-02-10T15:02:12.529708+0900 | compress | METRIC - time 0.27s
2026-02-10T15:02

(25/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.26it/s]

2026-02-10T15:03:26.852492+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.q_proj using 256 samples


2026-02-10T15:03:27.300738+0900 | compress | METRIC - time 0.45s
2026-02-10T15:03:27.301290+0900 | compress | METRIC - error 1349.91
2026-02-10T15:03:27.303881+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:03:27.304419+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:03:27.306398+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.k_proj using 256 samples
2026-02-10T15:03:27.566008+0900 | compress | METRIC - time 0.26s
2026-02-10T15:03:27.566451+0900 | compress | METRIC - error 358.52
2026-02-10T15:03:27.567437+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:03:27.567753+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:03:27.568686+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.v_proj using 256 samples
2026-02-10T15:03:27.825898+0900 | compress | METRIC - time 0.26s
2026-02-10T15:0

(26/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:00<00:00,  4.24it/s]

2026-02-10T15:04:42.559121+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.q_proj using 256 samples


2026-02-10T15:04:42.984579+0900 | compress | METRIC - time 0.43s
2026-02-10T15:04:42.985065+0900 | compress | METRIC - error 1540.80
2026-02-10T15:04:42.986200+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:04:42.986512+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:04:42.988196+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.k_proj using 256 samples
2026-02-10T15:04:43.248453+0900 | compress | METRIC - time 0.26s
2026-02-10T15:04:43.248887+0900 | compress | METRIC - error 389.04
2026-02-10T15:04:43.249808+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:04:43.250091+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:04:43.250967+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.v_proj using 256 samples
2026-02-10T15:04:43.510484+0900 | compress | METRIC - time 0.26s
2026-02-10T15:0

(27/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.28it/s]

2026-02-10T15:05:57.364276+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.q_proj using 256 samples


2026-02-10T15:05:57.833632+0900 | compress | METRIC - time 0.47s
2026-02-10T15:05:57.834248+0900 | compress | METRIC - error 1847.85
2026-02-10T15:05:57.837676+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:05:57.838406+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:05:57.840480+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.k_proj using 256 samples
2026-02-10T15:05:58.124079+0900 | compress | METRIC - time 0.28s
2026-02-10T15:05:58.124633+0900 | compress | METRIC - error 499.15
2026-02-10T15:05:58.125818+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:05:58.126141+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:05:58.127250+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.v_proj using 256 samples
2026-02-10T15:05:58.429345+0900 | compress | METRIC - time 0.30s
2026-02-10T15:0

(28/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:05<00:00,  3.91it/s]

2026-02-10T15:07:20.921129+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.q_proj using 256 samples


2026-02-10T15:07:21.393329+0900 | compress | METRIC - time 0.47s
2026-02-10T15:07:21.393862+0900 | compress | METRIC - error 2792.98
2026-02-10T15:07:21.396712+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:07:21.397244+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:07:21.399204+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.k_proj using 256 samples
2026-02-10T15:07:21.667155+0900 | compress | METRIC - time 0.27s
2026-02-10T15:07:21.667633+0900 | compress | METRIC - error 718.30
2026-02-10T15:07:21.668690+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:07:21.668997+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:07:21.669902+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.v_proj using 256 samples
2026-02-10T15:07:22.025461+0900 | compress | METRIC - time 0.36s
2026-02-10T15:0

(29/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [01:01<00:00,  4.16it/s]

2026-02-10T15:08:39.268205+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.q_proj using 256 samples


2026-02-10T15:08:39.710730+0900 | compress | METRIC - time 0.44s
2026-02-10T15:08:39.711364+0900 | compress | METRIC - error 3212.32
2026-02-10T15:08:39.713767+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:08:39.714115+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:08:39.716246+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.k_proj using 256 samples
2026-02-10T15:08:39.967586+0900 | compress | METRIC - time 0.25s
2026-02-10T15:08:39.967994+0900 | compress | METRIC - error 828.37
2026-02-10T15:08:39.969049+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:08:39.969346+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:08:39.970222+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.v_proj using 256 samples
2026-02-10T15:08:40.217876+0900 | compress | METRIC - time 0.25s
2026-02-10T15:0

(30/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:59<00:00,  4.27it/s]

2026-02-10T15:09:54.240882+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.q_proj using 256 samples


2026-02-10T15:09:54.666487+0900 | compress | METRIC - time 0.43s
2026-02-10T15:09:54.666978+0900 | compress | METRIC - error 3182.92
2026-02-10T15:09:54.668476+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:09:54.668833+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-10T15:09:54.670752+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.k_proj using 256 samples
2026-02-10T15:09:54.938559+0900 | compress | METRIC - time 0.27s
2026-02-10T15:09:54.939351+0900 | compress | METRIC - error 902.42
2026-02-10T15:09:54.940503+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-10T15:09:54.940844+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-10T15:09:54.941778+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.v_proj using 256 samples
2026-02-10T15:09:55.205383+0900 | compress | METRIC - time 0.26s
2026-02-10T15:0

(31/31): Propagating: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:00<00:00, 3423.77it/s]

2026-02-10T15:10:09.784815+0900 | finalize | INFO - Compression lifecycle finalized for 1 modifiers
2026-02-10T15:10:09.792343+0900 | post_process | WARNING - Optimized model is not saved. To save, please provide`output_dir` as input arg.Ex. `oneshot(..., output_dir=...)`



[INFO] GPTQ 양자화 완료!


# 6. 모델 저장

In [6]:
print("[INFO] 모델 저장 중...")

# 기존 폴더 삭제
if os.path.exists(OUT_DIR):
    shutil.rmtree(OUT_DIR)
os.makedirs(OUT_DIR, exist_ok=True)

model.save_pretrained(OUT_DIR, save_compressed=True)
tokenizer.save_pretrained(OUT_DIR)

# 저장된 파일 확인
print(f"\n[INFO] 저장된 파일:")
total_size = 0
for f in sorted(os.listdir(OUT_DIR)):
    size = os.path.getsize(os.path.join(OUT_DIR, f))
    total_size += size
    print(f"  {f}: {size/1e6:.1f} MB")

quantized_size_gb = total_size / 1e9

print("\n" + "=" * 60)
print("모델 크기 비교")
print("=" * 60)
print(f"  원본 모델:     {ORIGINAL_MODEL_SIZE_GB:.2f} GB")
print(f"  양자화 모델:   {quantized_size_gb:.2f} GB")
print(f"  압축률:        {quantized_size_gb / ORIGINAL_MODEL_SIZE_GB * 100:.1f}%")
print("=" * 60)

[INFO] 모델 저장 중...
2026-02-10T15:10:09.835676+0900 | get_model_compressor | INFO - skip_sparsity_compression_stats set to True. Skipping sparsity compression statistic calculations. No sparsity compressor will be applied.


Compressing model: 210it [00:03, 61.68it/s]



[INFO] 저장된 파일:
  chat_template.jinja: 0.0 MB
  config.json: 0.0 MB
  generation_config.json: 0.0 MB
  merges.txt: 1.2 MB
  model.safetensors: 1407.7 MB
  recipe.yaml: 0.0 MB
  special_tokens_map.json: 0.0 MB
  tokenizer.json: 7.9 MB
  tokenizer_config.json: 0.1 MB
  vocab.json: 1.9 MB

모델 크기 비교
  원본 모델:     2.56 GB
  양자화 모델:   1.42 GB
  압축률:        55.4%


# 7. 제출 파일 생성

In [7]:
zip_name = "submit_optimized"
print(f"[INFO] {zip_name}.zip 생성 중...")

# 기존 zip 삭제
if os.path.exists(f"{zip_name}.zip"):
    os.remove(f"{zip_name}.zip")

shutil.make_archive(
    base_name=zip_name,
    format="zip",
    root_dir=".",
    base_dir=OUT_DIR,
)

zip_size = os.path.getsize(f"{zip_name}.zip") / 1e9
print(f"[INFO] 생성 완료: {zip_name}.zip ({zip_size:.2f} GB)")

if zip_size <= 10:
    print("✅ 용량 제한 충족 (≤ 10GB)")
else:
    print("❌ 용량 초과!")

# 제출 구조 확인
print("\n" + "=" * 60)
print("제출 파일 구조")
print("=" * 60)
print(f"{zip_name}.zip")
print(f"└── model/")
for f in sorted(os.listdir(OUT_DIR))[:5]:
    print(f"    ├── {f}")
print("    └── ...")
print("=" * 60)

[INFO] submit_optimized.zip 생성 중...
[INFO] 생성 완료: submit_optimized.zip (0.88 GB)
✅ 용량 제한 충족 (≤ 10GB)

제출 파일 구조
submit_optimized.zip
└── model/
    ├── chat_template.jinja
    ├── config.json
    ├── generation_config.json
    ├── merges.txt
    ├── model.safetensors
    └── ...


---

# 예상 성능 개선

## 평가 공식
```
Score = 0.5 × PerfNorm + 0.5 × SpeedNorm
```

## 최적화 효과 분석

| 항목 | 베이스라인 | 최적화 버전 | 변화 |
|------|-----------|------------|------|
| **PerfNorm** | ~0.95 | ~0.97 | ↑ actorder="weight" |
| **SpeedNorm** | ~0.30 | ~0.30 | = (동일 압축률) |
| **Score** | ~0.625 | **~0.635** | ↑ |

## 추가 개선 방안

### 1. 캘리브레이션 품질 향상
```python
NUM_CALIBRATION_SAMPLES = 512  # 256 → 512
MAX_SEQUENCE_LENGTH = 1024     # 512 → 1024
```

### 2. AWQ 시도 (Marlin-AWQ가 가장 빠름)
- vLLM 벤치마크: AWQ 741 tok/s vs GPTQ 712 tok/s

---

## 참고 자료
- [vLLM Quantization Benchmarks](https://docs.jarvislabs.ai/blog/vllm-quantization-complete-guide-benchmarks)
- [LLM Compressor Docs](https://docs.vllm.ai/projects/llm-compressor/en/stable/)
- [GPTQModel GitHub](https://github.com/ModelCloud/GPTQModel)

---